src/runtime_src/core/edge/ps_kernels/xrt/tests/validate/ps_bandwidth_test/ps_bandwidth.cpp

```diff
@@ -0,0 +1,101 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <syslog.h>
+#include <sys/mman.h>
+
+#include "xrt/xrt_kernel.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#include "core/edge/include/sk_types.h"
+
+// User private data structure container (context object) definition
+class xrtHandles : public pscontext
+{
+public:
+  xrt::device dhdl;
+  xrt::kernel bandwidth_kernel;
+  xrtHandles(xclDeviceHandle dhdl_in, const xuid_t xclbin_uuid)
+    : dhdl(dhdl_in)
+    , bandwidth_kernel(dhdl,xclbin_uuid,"bandwidth")
+  {
+  }
+};
+
+  __attribute__((visibility("default")))
+  pscontext *bandwidth_kernel_init(xclDeviceHandle dhdl, const xuid_t xclbin_uuid) {
+    xrtHandles *handles = new xrtHandles(dhdl, xclbin_uuid);
+
+    return(handles);
+  }
+  
+__attribute__((visibility("default")))
+int bandwidth_kernel(int reps, double *max_throughput, xrtHandles *xrtHandle)
+{
+  double max_throughput_int = 0;
+
+  // Starting at 4K and going up to 16M with increments of power of 2
+  for (uint32_t i = 4 * 1024; i <= 16 * 1024 * 1024; i *= 2) {
```

Comment:
Thank you for the comment, without it we wouldn't really know the purpose behind this for loop.  That said it, this line should be refactored to something like this:
```
constexpr uint32_t 4K = 4 * 1024;
constexpr uint32_t 16M = 15 * 1024 * 1024;
for (uint32_t i = 4K; i <= 16M; i *= 2)

```
And yes, I know that 4K and 16M are not valid variable names, but you get the idea.

Reply:
I guess someone came with a constexpr UDL library so you can write 4_Ki and 16_Mi https://en.cppreference.com/w/cpp/language/user_literal